project/cmake/scripts/common/Macros.cmake

```diff
@@ -367,13 +367,32 @@ function(core_find_git_rev)
   endif()
 endfunction()
 
+# Parses version.txt and libKODI_guilib.h and sets variables
+# used to construct dirs structure, file naming, API versions, etc.
+#
+# The following variables are set from version.txt:
+#   APP_NAME - app name
+#   APP_NAME_LC - lowercased app name
+#   APP_NAME_UC - uppercased app name
+#   COMPANY_NAME - company name
+#   APP_VERSION_MAJOR - the app version major
+#   APP_VERSION_MINOR - the app version minor
+#   APP_VERSION_TAG - the app version tag
+#   APP_VERSION_TAG_LC - lowercased app version tag
+#   APP_VERSION - the app version (${APP_VERSION_MAJOR}.${APP_VERSION_MINOR}-${APP_VERSION_TAG})
+#   APP_ADDON_API - the addon API version in the form of 16.9.702
+#   FILE_VERSION - file version in the form of 16,9,702, Windows only
+#
+# The following variables are set from libKODI_guilib.h:
+#   guilib_version - current ADDONGUI API version
+#   guilib_version_min - minimal ADDONGUI API version
 macro(core_find_versions)
   include(CMakeParseArguments)
   core_file_read_filtered(version_list ${CORE_SOURCE_DIR}/version.txt)
   string(REPLACE " " ";" version_list "${version_list}")
```

Comment:
unrelated, but since you're at it: since the macro will put all vars in the parent scope, mind using unset(version_list) as that one is local. 

(hope these kind of requests are ok, but it's probably less effort to include in that commit instead of starting a new one)

Nice work with the docs, btw!